tools/istio-iptables/pkg/cmd/run.go

```diff
@@ -390,7 +400,10 @@ func (iptConfigurator *IptablesConfigurator) run() {
 
 		// Do not redirect app calls to back itself via Envoy when using the endpoint address
 		// e.g. appN => appN by lo
-		iptConfigurator.iptables.AppendRuleV4(constants.ISTIOOUTPUT, constants.NAT, "-o", "lo", "-m", "owner", "!", "--gid-owner", gid, "-j", constants.RETURN)
+		// If loopback explicitly set via OutboundIPRangesInclude, then don't return.
+		if !ipv4RangesInclude.HasLoopBackIP && !ipv6RangesInclude.HasLoopBackIP {
+			iptConfigurator.iptables.AppendRuleV4(constants.ISTIOOUTPUT, constants.NAT, "-o", "lo", "-m", "owner", "!", "--gid-owner", gid, "-j", constants.RETURN)
+		}
```

Comment:
You have to make same change in cni as well.

Reply:
I will send out a PR for CNI.